runtime/jvmti/suspendhelper.cpp

```diff
@@ -39,33 +39,49 @@ suspendThread(J9VMThread *currentThread, jthread thread, BOOLEAN allowNull, BOOL
 	}
 	rc = getVMThread(currentThread, thread, &targetThread, JVMTI_ERROR_NONE, flags);
 	if (rc == JVMTI_ERROR_NONE) {
-		if (targetThread->publicFlags & J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND) {
-			rc = JVMTI_ERROR_THREAD_SUSPENDED;
-		} else {
-			if (targetThread->publicFlags & J9_PUBLIC_FLAGS_STOPPED) {
-				/* Do not suspend dead threads. Mirrors SUN behaviour. */
-				rc = JVMTI_ERROR_THREAD_NOT_ALIVE;
+		J9JavaVM *vm = currentThread->javaVM;
+#if JAVA_SPEC_VERSION >= 19
+		j9object_t threadObject = (NULL == thread) ? currentThread->threadObject : J9_JNI_UNWRAP_REFERENCE(thread);
+		if (NULL != targetThread)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		{
+			if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
+				rc = JVMTI_ERROR_THREAD_SUSPENDED;
 			} else {
-				if (currentThread == targetThread) {
-					*currentThreadSuspended = TRUE;
+				if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_STOPPED)) {
+					/* Do not suspend dead threads. Mirrors SUN behaviour. */
+					rc = JVMTI_ERROR_THREAD_NOT_ALIVE;
```

Comment:
Have you verified this behaviour with the normal thread suspend? I don't see anything in the new JVMTI spec calling this out (this error is not listed as a possible return code).

Reply:
Sorry, confused by the diff again - this only applies to real threads.